enhancements/cert-manager/istio-csr-controller.md

```diff
@@ -178,182 +205,271 @@ type IstioCSR struct {
 
-	// Affinity is for setting scheduling affinity rules.
+	// affinity is for setting scheduling affinity rules.
 	// ref: https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/
+	// +kubebuilder:validation:Optional
 	// +optional
 	Affinity *corev1.Affinity `json:"affinity,omitempty"`
 
-	// Tolerations is for setting the pod tolerations.
+	// tolerations is for setting the pod tolerations.
 	// ref: https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/
+	// +listType=atomic
+	// +kubebuilder:validation:MinItems:=0
+	// +kubebuilder:validation:MaxItems:=10
+	// +kubebuilder:validation:Optional
 	// +optional
 	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
 
-	// NodeSelector is for defining the scheduling criteria using node labels.
+	// nodeSelector is for defining the scheduling criteria using node labels.
 	// ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
+	// +mapType=atomic
+	// +kubebuilder:validation:MinProperties:=0
+	// +kubebuilder:validation:MaxProperties:=10
+	// +kubebuilder:validation:Optional
 	// +optional
 	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
 }
 
 // CertManagerConfig is for configuring cert-manager specifics.
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.issuerRef) && !has(self.issuerRef) || has(oldSelf.issuerRef) && has(self.issuerRef)",message="issuerRef may only be configured during creation"
 type CertManagerConfig struct {
 	// issuerRef contains details to the referenced object used for
-	// obtaining the certificates.
-	// When unset operator will create Issuer in the configured istio's
-	// namespace.
-	// +kubebuilder:validation:XValidation:rule="oldSelf == '' || self == oldSelf",message="issuerRef is immutable once set"
+	// obtaining the certificates. When issuerRef.Kind is Issuer, it must exist in the
+	// .spec.istioCSRConfig.istio.namespace.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="issuerRef is immutable once set"
+	// +kubebuilder:validation:XValidation:rule="self.kind.lowerAscii() == 'issuer' || self.kind.lowerAscii() == 'clusterissuer'",message="kind must be either 'Issuer' or 'ClusterIssuer'"
+	// +kubebuilder:validation:XValidation:rule="self.group.lowerAscii() == 'cert-manager.io'",message="group must be 'cert-manager.io'"
+	// +kubebuilder:validation:Required
 	// +required
-	IssuerRef certmanagerv1.ObjectReference `json:"issuerRef,omitempty"`
+	IssuerRef *certmanagerv1.ObjectReference `json:"issuerRef"`
+
+	// istioCACertificate when provided, the operator will use the CA certificate from the specified ConfigMap. If empty, the operator will
+	// automatically extract the CA certificate from the secret containing the istiod certificate obtained from cert-manager.
+	// +kubebuilder:validation:Optional
+	// +optional
+	IstioCACertificate *ConfigMapReference `json:"istioCACertificate,omitempty"`
```

Comment:
We can call out the validations in place as well for better UX: ike configMap should exist, the configured key should be present, and the key should be PEM content and a CA chain.

Reply:
I feel these are implicit.